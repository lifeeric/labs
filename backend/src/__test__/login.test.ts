import * as util from "util";
import axios from "axios";
import { exec } from "child_process";
import path from "path";

// const pExec = util.promisify(exec);

const API = "http://localhost:8000/graphql";
// const CMD_SEED_DB = path.join(__dirname, "/../../dataset.sh");

// Invoke axios
const axios_post = async (API: string, query: string) => {
  return await axios.post(API, {
    query,
  });
};

describe("Getting login user", () => {
  beforeAll(async () => {
    await pExec(CMD_SEED_DB);
  });

  it("should retrieves the user data", async () => {
    const query = `
        query {
            login(email:"alex@test.com", password: "alex"){
              __typename
              ... on LoginUserResult {
                email
              }
              
            ... on Error {
              error
              message
            }
            }
          }
        `;

    const response = await axios_post(API, query);
    expect(response.data).toMatchSnapshot();
  });
});
