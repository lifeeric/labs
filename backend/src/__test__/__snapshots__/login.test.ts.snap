// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Getting login user should retrieves the user data 1`] = `
Object {
  "data": Object {
    "login": Object {
      "__typename": "LoginUserResult",
      "email": "alex@test.com",
    },
  },
}
`;
